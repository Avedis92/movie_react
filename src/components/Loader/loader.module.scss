@import "../sasscommonmixxins/flex.scss";
.loader {
  width: 10vw;
  height: 10vh;
  /*margin-left: 45vw;
  margin-top: 20vh;*/
  padding-top: 15%;
  padding-left: 45vw;
  padding-bottom: 5%;
  @include flexElements(flex, row, center, space-evenly);
}
.loader > span {
  background: black;
  border-radius: 50%;
  animation: bouncingBalls 0.6s infinite alternate;
}
.loader > span:nth-child(2) {
  animation-delay: 0.2s;
}

.loader > span:nth-child(3) {
  animation-delay: 0.4s;
}

@keyframes bouncingBalls {
  from {
    width: 10px;
    height: 10px;
    opacity: 1;
    transform: translate3d(0);
  }
  to {
    width: 20px;
    height: 20px;
    opacity: 0.3;
    transform: translate3d(0, -20px, 0);
  }
}
