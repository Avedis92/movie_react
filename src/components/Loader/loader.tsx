import styles from './loader.module.scss';

function Loader() {
  return (
    <div className={styles.loader}>
      <span />
      <span />
      <span />
    </div>
  );
}

export default Loader;
