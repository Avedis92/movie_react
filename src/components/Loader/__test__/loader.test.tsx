import { render } from '@testing-library/react';
import Loader from '../loader';

test('Create a snapshot test', () => {
  const {asFragment} = render(<Loader />);
  expect(asFragment()).toMatchSnapshot();
});
