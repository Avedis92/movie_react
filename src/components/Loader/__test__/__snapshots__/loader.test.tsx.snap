// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="loader"
  >
    <span />
    <span />
    <span />
  </div>
</DocumentFragment>
`;
